src/viking.js

```diff
@@ -1,11 +1,74 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health;
+    this.strength = strength;
+    this.attack = function (){
+        return this.strength;
+    };
+    this.receiveDamage = function (damage) {
+        this.health -= damage; 
+    }
+}
 
```

Comment:
Here you're adding the Soldier methods (attack and receiveDamage) through the Soldier constructor itself.
Nothing wrong with that, but it's a bit inefficient as every object you will instance from now on (Viking and Sakon for ex) is going to have those methods inside.
To avoid duplicating them to each and every object instance, you can add those methods to the Soldier prototype, and Viking and Saxon will be able to borrow the methods they need from their parent (Soldier).